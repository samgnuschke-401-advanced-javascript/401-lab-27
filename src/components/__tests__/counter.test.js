import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../counter/counter';

describe("<Counter />", () => {
  it('is rendered at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('.downClicker').exists()).toBe(true);
    expect(app.find('.upClicker').exists()).toBe(true);
    expect(app.find('.counter').exists()).toBe(true);
  });

  it('subtracts 1 when clicked', () => {
    const app = mount(<Counter />);
    const downClicker = app.find('.downClicker');
    downClicker.simulate('click');
    expect(app.state('count')).toEqual(-1);
  })

  it('adds 1 when clicked', () => {
    const app = mount(<Counter />);
    const upClicker = app.find('.upClicker');
    upClicker.simulate('click');
    expect(app.state('count')).toEqual(1);
  })

  it('snapshot will be rendered', () => {
    const count = renderer
      .create(<Counter />)
      .toJSON();
    expect(count).toMatchSnapshot();
  });
});