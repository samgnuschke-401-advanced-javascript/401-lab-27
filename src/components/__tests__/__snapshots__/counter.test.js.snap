// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> snapshot will be rendered 1`] = `
<section
  className="counter"
>
  <a
    className="downClicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upClicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
